mm2src/coins/nft.rs

```diff
@@ -306,13 +292,72 @@ pub async fn update_nft(ctx: MmArc, req: UpdateNftReq) -> MmResult<(), UpdateNft
             &req.url_antispam,
         )
         .await?;
+        update_nft_global_in_coins_ctx(&ctx, &storage, *chain, nft_global).await?;
         update_transfers_with_empty_meta(&storage, chain, &req.url, &req.url_antispam).await?;
         update_spam(&storage, *chain, &req.url_antispam).await?;
         update_phishing(&storage, chain, &req.url_antispam).await?;
     }
     Ok(())
 }
 
+/// Updates the global NFT information in the coins context.
+///
+/// This function uses the up-to-date NFT list for a given chain and updates the
+/// corresponding global NFT information in the coins context.
+async fn update_nft_global_in_coins_ctx<T>(
+    ctx: &MmArc,
+    storage: &T,
+    chain: Chain,
+    mut nft_global: EthCoin,
+) -> MmResult<(), UpdateNftError>
+where
+    T: NftListStorageOps + NftTransferHistoryStorageOps,
+{
+    let nft_list = storage.get_nft_list(vec![chain], true, 1, None, None).await?;
+    let coins_ctx = CoinsContext::from_ctx(ctx).map_to_mm(UpdateNftError::Internal)?;
+    let mut coins = coins_ctx.coins.lock().await;
+
+    update_nft_infos(&mut nft_global, nft_list.nfts).await;
+
+    coins.insert(
+        nft_global.ticker.to_string(),
+        MmCoinStruct::new(MmCoinEnum::EthCoin(nft_global)),
+    );
```

Comment:
I don't understand why it's done this way, isn't nft infos already an Arc that can be shared among threads https://github.com/KomodoPlatform/komodo-defi-framework/blob/11284b94ded5f3a1ed9cb9a1009d8d0094a394a0/mm2src/coins/eth.rs#L466

Reply:
Thanks for the note!
You are right, we can access coins context ones and update `nfts_infos` directly.
Updated `update_nft_global_in_coins_ctx` function.